maestro-cli/src/main/java/maestro/cli/report/HtmlTestSuiteReporter.kt

```diff
@@ -0,0 +1,108 @@
+package maestro.cli.report
+
+import maestro.cli.model.TestExecutionSummary
+import okio.Sink
+import okio.buffer
+import org.slf4j.LoggerFactory
+import java.io.BufferedReader
+import java.io.FileReader
+import maestro.cli.report.TestDebugReporter
+
+class HtmlTestSuiteReporter : TestSuiteReporter {
+    override fun report(summary: TestExecutionSummary, out: Sink) {
+        val bufferedOut = out.buffer()
+        val htmlContent = buildHtmlReport(summary)
+        bufferedOut.writeUtf8(htmlContent)
+        bufferedOut.close()
+    }
+
+    private fun getTestStep(): Array<Array<String>> {
```

Comment:
Parsing out the test steps this way feels brittle and will be hard to maintain and break if the output of the test execution changes, so let's make sure that there is no dependency between the generated log and the html report. When running this locally I can't get it to work due to the parsing failing:

```
3/4 Flows Failed

java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at maestro.cli.report.HtmlTestSuiteReporter.buildHtmlReport(HtmlTestSuiteReporter.kt:95)
```

I believe the way forward here is either to
- Update `TestExecutionSummary.FlowResult` to contain information about each step
- Remove the steps from the report for now and follow up with this in a separate PR